core/src/main/kotlin/event/Event.kt

```diff
@@ -45,3 +51,17 @@ public interface Event : KordObject {
     @KordPreview
     public val customContext: Any?
 }
+
+/**
+ * Representation of an event that has not been documented / added to Kord yet.
+ *
+ * @property data the raw [json body][JsonElement] of this event if provided
+ * @property name the name of the event if provided
+ */
+public data class UnknownEvent(
```

Comment:
other events aren't data classes, `equals`, `hashCode` and `copy` don't make much sense for events
i think we should instead just provide a custom `toString` implementation